### PR TITLE
[fix](log) column index off-by-one error in scanner logs

### PR DESCRIPTION
**Pull request by kaijchen**
# Proposed changes

Issue Number: close #xxx

## Problem summary

The following log is printed when using strict mode:

```
"column({col[1]}) value is incorrect while strict mode is true, src value is {src[2]}",
```

Which should be:

```
"column({col[1]}) value is incorrect while strict mode is true, src value is {src[1]}",
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/base_scanner.cpp [hunk 1]
```diff
@@ -221,7 +221,7 @@ Status BaseScanner::_materialize_dest_block(vectorized::Block* dest_block) {
         if (slot_desc->type().is_variant_type()) {
             continue;
         }
-        int dest_index = ctx_idx++;
+        int dest_index = ctx_idx;
```
Comment (kaijchen) [inline be/src/exec/base_scanner.cpp:224]: `ctx_idx` is referenced at line 254.

File: be/src/vec/exec/scan/vfile_scanner.cpp [hunk 2]
```diff
@@ -473,7 +473,7 @@ Status VFileScanner::_convert_to_output_block(Block* block) {
         if (!slot_desc->is_materialized()) {
             continue;
         }
-        int dest_index = ctx_idx++;
+        int dest_index = ctx_idx;
```
Comment (kaijchen) [inline be/src/vec/exec/scan/vfile_scanner.cpp:476]: `ctx_idx` is referenced at line 504.

**Comment by kaijchen** [comment]
Cc @dataroaring, PTAL thanks!

**Comment by morningman** [comment]
Please describe this pr

**Comment by kaijchen** [comment]
> Please describe this pr

Updated.

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by HHoflittlefish777** [comment]
LGTM

**Review by sollhui — APPROVED** [review]
LGTM

**Comment by kaijchen** [comment]
run clickbench

**Comment by kaijchen** [comment]
run clickbench-new

**Comment by kaijchen** [comment]
run approval

**Comment by kaijchen** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.92 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17156311194 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.55 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17170493721 Bytes

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**